src/components/BookInfoCol.js

```diff
@@ -1,25 +1,34 @@
 import { useSelector, useDispatch } from 'react-redux';
 import PropTypes from 'prop-types';
+import { deleteBook, fetchBooks } from '../redux/books/booksSlice';
 import infoCss from './info.module.css';
-import { removeBook } from '../redux/books/booksSlice';
 
 const BookInfo = ({ bookId }) => {
-  const book = useSelector((state) => state.books.find((book) => book.item_id === bookId));
   const dispatch = useDispatch();
-  const deleteBook = (id) => {
-    dispatch(removeBook({ item_id: id }));
-  };
 
+  const book = useSelector((state) => Object.entries(state.books.data)
+    .find((book) => bookId === book[0]));
+
+  const deleteABook = async (bookId) => {
+    await dispatch(deleteBook({ bookId }));
+    dispatch(fetchBooks());
```

Comment:
**OPTIONAL** - I suggest calling this function in your `removeBook` action. Please fix this 🙏. You can check this [article](https://stackoverflow.com/questions/68871562/dispatch-action-on-the-createasyncthunk) for how to dispatch action inside a `createAsyncThunk` function 👍 👍

Reply:
I've made the optional changes. Thanks for the heads up